exporter/splunkhecexporter/client.go

```diff
@@ -356,11 +407,17 @@ func (c *client) pushMetricsDataInBatches(ctx context.Context, md pmetric.Metric
 
 	for !is.done {
 		buf.Reset()
-		latestIterState, batchPermanentErrors := c.fillMetricsBuffer(md, buf, is)
+		var latestIterState iterState
+		var batchPermanentErrors []error
+		if c.config.UseMultiMetricFormat {
+			latestIterState, batchPermanentErrors = c.fillMetricsBufferMultiMetrics(md, buf, is)
+		} else {
+			latestIterState, batchPermanentErrors = c.fillMetricsBuffer(md, buf, is)
+		}
 		permanentErrors = append(permanentErrors, batchPermanentErrors...)
 		if !buf.Empty() {
 			if err := c.postEvents(ctx, buf, headers); err != nil {
-				return consumererror.NewMetrics(err, subMetrics(md, is))
+				return consumererror.NewMetrics(err, md)
```

Comment:
Removing `subMetrics` will cause double publishing if one batch is accepted and another one is rejected by the backend. It'll be hard to do for the merged events, so it probably can be ignored. But we are introducing a bug for the default behavior (UseMultiMetricFormat=false)

Reply:
Yes, it's a change, is it going too far or is it acceptable?

Reply:
I went ahead and reverted.